Redefinition of strlcpy leads to a compilation error

On Debian Trixie, the compilation is broken since commit bc5ae9b:

https://github.com/lxc/incus/blob/819de904b3ae0ebf9f826d26728a19ce237e8e6e/shared/cgo/process_utils.h#L117-L128

```
CC="cc" CGO_LDFLAGS_ALLOW="(-Wl,-wrap,pthread_create)|(-Wl,-z,now)" go install -v -tags "libsqlite3"  ./...
github.com/lxc/incus/v6/shared/idmap
github.com/lxc/incus/v6/internal/linux
# github.com/lxc/incus/v6/internal/linux
In file included from internal/linux/socket_linux_cgo.go:25:
./../../shared/cgo/process_utils.h:117:22: error: static declaration of 'strlcpy' follows non-static declaration
  117 | static inline size_t strlcpy(char *dest, const char *src, size_t size)
      |                      ^~~~~~~
In file included from internal/linux/socket_linux_cgo.go:18:
/usr/include/string.h:506:15: note: previous declaration of 'strlcpy' with type 'size_t(char * restrict,  const char * restrict,  size_t)' {aka 'long unsigned int(char * restrict,  const char * restrict,  long unsigned int)'}
  506 | extern size_t strlcpy (char *__restrict __dest,
      |               ^~~~~~~
# github.com/lxc/incus/v6/shared/idmap
In file included from shared/idmap/shift_linux.go:26:
./../../shared/cgo/process_utils.h:117:22: error: static declaration of 'strlcpy' follows non-static declaration
  117 | static inline size_t strlcpy(char *dest, const char *src, size_t size)
      |                      ^~~~~~~
In file included from shared/idmap/shift_linux.go:17:
/usr/include/string.h:506:15: note: previous declaration of 'strlcpy' with type 'size_t(char * restrict,  const char * restrict,  size_t)' {aka 'long unsigned int(char * restrict,  const char * restrict,  long unsigned int)'}
  506 | extern size_t strlcpy (char *__restrict __dest,
      |               ^~~~~~~
make: *** [Makefile:37: build] Error 1
```

--- a/shared/cgo/process_utils.h
+++ b/shared/cgo/process_utils.h
@@ -114,19 +114,6 @@ static inline int push_vargs(char ***list, char *entry)
 	return 0;
 }
 
-static inline size_t strlcpy(char *dest, const char *src, size_t size)
-{
-	size_t ret = strlen(src);
-
-	if (size) {
-		size_t len = (ret >= size) ? size - 1 : ret;
-		memcpy(dest, src, len);
-		dest[len] = '\0';
-	}
-
-	return ret;
-}
-
 /*
  * Sets the process title to the specified title. Note that this may fail if
  * the kernel doesn't support PR_SET_MM_MAP (kernels <3.18).
@@ -231,8 +218,11 @@ static inline int setproctitle(char *title)
 
 	ret = prctl(PR_SET_MM, prctl_arg(PR_SET_MM_MAP), prctl_arg(&prctl_map),
 		    prctl_arg(sizeof(prctl_map)), prctl_arg(0));
-	if (ret == 0)
-		(void)strlcpy((char *)arg_start, title, len);
+	if (ret == 0) {
+		char *dest = (char *)arg_start;
+		memcpy(dest, title, len - 1);
+		dest[len-1] = '\0';
+	}
 
 	return ret;
 }

